### PR TITLE
feat(#66): Add filter for feedback listing page in feedback-spa

### PR DESCRIPTION
**Pull request by anjneeksharma**
Added filter on feedback listing page user can filter feedback based on openen/closed status and based on module.

## Does this PR introduce a breaking change?

No

## Screenshots:

image

### Ready-for-merge Checklist

- [x ] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)?
- [x ] Does the change have appropriate unit tests?
- [x ] Did tests pass?
- [x ] Did you update or add any necessary documentation (README.md, WHY.md, etc.)?
- [ x] Was this feature demo'd and the design review approved?

### CODE REVIEW

File: packages/feedback-spa/src/app/modules/feedback/home/feedback-home.component.ts [hunk 1]
```diff
@@ -13,6 +13,50 @@ export class FeedbackHomeComponent implements OnInit {
...
+      name: 'Feedback'
+    },
+    {
+      id: 4,
+      name: 'Projects & Initiatives'
+    },
+    {
+      id: 5,
+      name: 'Service Status'
+    }
```
Comment (deshmukhmayur) [inline packages/feedback-spa/src/app/modules/feedback/home/feedback-home.component.ts:54]: Why do we have these options here? These modules are not in the Platform, and also are not a part of the MVP.
Make these dynamic, and use a mock API call to fetch these.

File: packages/feedback-service/dist/bundle.js [hunk 2]
```diff
@@ -0,0 +1 @@
...D: ID!): FeedbackType\n}\n\ntype Mutation {\n  # Mutation Definitions\n  create(input: FeedbackInput): FeedbackType\n  update(input: FeedbackInput): FeedbackType\n  delete(_id: String!): FeedbackType\n}\n\ntype Subscription {\n  # Subscription Definitions\n  updatedFeedbackData: FeedbackType\n}\n\ntype FeedbackType {\n  message: String\n}\n\ninput FeedbackInput {\n  message: String\n}\n\n",name:"GraphQL request",locationOffset:{line:1,column:1}};e.exports=i},function(e,n,i){"use strict";Object.defineProperty(n,"__esModule",{value:!0});const t=i(11);n.FeedbackResolver={Subscription:{updatedFeedbackData:{subscribe:()=>t.pubsub.asyncIterator("Feedback_UPDATE")}},Query:{list:(e,n,i)=>[{message:"GET API for Feedback microservice"}],get:(e,n,i)=>({message:"GET by ID API for Feedback microservice"})},Mutation:{create:(e,n,i)=>({message:"POST API for Feedback microservice"}),update:(e,n,i)=>({message:"PUT API for Feedback microservice"}),delete:(e,n,i)=>({message:"DELETE API for Feedback microservice"})}}},function(e,n,i){"use strict";Object.defineProperty(n,"__esModule",{value:!0});const t=i(12);n.pubsub=new t.RedisPubSub},function(e,n){e.exports=require("graphql-redis-subscriptions")}]);
```
Comment (deshmukhmayur) [inline packages/feedback-service/dist/bundle.js:1]: Why is this file here? This should've been ignored by git.

File: packages/feedback-spa/src/app/modules/feedback/dropdown/dropdown.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { DropdownComponent } from './dropdown.component';
+
+describe('DropdownComponent', () => {
+  let component: DropdownComponent;
+  let fixture: ComponentFixture<DropdownComponent>;
+
+  beforeEach(async(() => {
```
[the remark this answers is not in the document]
Comment (anjneeksharma) [reply packages/feedback-spa/src/app/modules/feedback/dropdown/dropdown.component.spec.ts]: If we create any component in Angular 8 this code automatically generates i neither copy it nor write it myself.

File: packages/feedback-spa/src/app/modules/feedback/dropdown/dropdown.component.ts [hunk 4]
```diff
@@ -26,7 +24,7 @@ export class DropdownComponent implements OnInit {
   @Input() truncateLength?: number;
   @Input() disableAny: boolean;
   // tslint:disable-next-line:no-output-on-prefix
```
Comment (deshmukhmayur) [inline packages/feedback-spa/src/app/modules/feedback/dropdown/dropdown.component.ts:26]: This line is not needed now, as you have removed `on` prefix from the selected property below this.

File: packages/feedback-spa/src/mocks/options.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  },
+  {
+    id: 4,
+    name: 'Projects & Initiatives'
+  },
+  {
+    id: 5,
+    name: 'Service Status'
+  }
+];
```
Comment (deshmukhmayur) [inline packages/feedback-spa/src/mocks/options.js:26]: Newline at the end of file.

```suggestion
];

```

File: packages/feedback-spa/src/mocks/options.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    name: 'Feedback'
+  },
+  {
+    id: 4,
+    name: 'Projects & Initiatives'
+  },
+  {
+    id: 5,
+    name: 'Service Status'
+  }
```
Comment (riginoommen) [inline packages/feedback-spa/src/mocks/options.js:25]: Please make sure that these options are valid from now onwards

Comment (anjneeksharma) [reply packages/feedback-spa/src/mocks/options.js:25]: @riginoommen  file was js i converted that inot .ts   is that your mean ?

Comment (riginoommen) [reply packages/feedback-spa/src/mocks/options.js]: i meant about the options like Service status, Projects & Inititives like that. Ensure that data that you push is valid

Comment (anjneeksharma) [reply packages/feedback-spa/src/mocks/options.js]: I discussed same with Sayak as he informed right now it will be like this latter i will be dynamic data instead of hardcore, for now its fine.

File: packages/feedback-spa/package.json [hunk 7]
```diff
@@ -43,7 +43,7 @@
   },
   "devDependencies": {
     "@angular-builders/jest": "9.0.1",
-    "@angular-devkit/build-angular": "0.901.0",
+    "@angular-devkit/build-angular": "^0.803.24",
```
Comment (riginoommen) [inline packages/feedback-spa/package.json:46]: Why this is downgraded. any reason.

Comment (anjneeksharma) [reply packages/feedback-spa/package.json:46]: with Angular 8 this is the perfect version we should use.

File: packages/feedback-spa/src/app/modules/feedback/dropdown/dropdown.component.ts (hunk 4 above)
Comment (anjneeksharma) [reply packages/feedback-spa/src/app/modules/feedback/dropdown/dropdown.component.ts]: Removed.

File: packages/feedback-spa/src/app/modules/feedback/home/feedback-home.component.ts (hunk 1 above)
Comment (anjneeksharma) [reply packages/feedback-spa/src/app/modules/feedback/home/feedback-home.component.ts]: As discussed these option will come dynamically for now i have created mock file.

File: packages/feedback-spa/src/app/modules/feedback/dropdown/dropdown.component.spec.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (deshmukhmayur) [reply packages/feedback-spa/src/app/modules/feedback/dropdown/dropdown.component.spec.ts]: No it's not. See the spec file for `app.component.spec.ts`. The async is useless in this case, so it shouldn't be there.

File: packages/feedback-spa/package.json (hunk 7 above)
Comment (deshmukhmayur) [reply packages/feedback-spa/package.json:46]: But is the `0.901` version causing any problems?

Also, any specific reason this app is built in Angular 8 and not 9?

Comment (anjneeksharma) [reply packages/feedback-spa/package.json:46]: Yes 0.901 version was causing an error so i removed it.
I chose Angular 8 because its LTS version.

File: packages/feedback-spa/src/app/modules/feedback/home/feedback-home.component.ts [hunk 8]
```diff
@@ -1,6 +1,7 @@
 import { Component, OnInit, ViewChild, ElementRef } from '@angular/core';
 import { FeedbackService } from '../feedback.service';
```
Comment (deshmukhmayur) [inline packages/feedback-spa/src/app/modules/feedback/home/feedback-home.component.ts:2]: Wait a minute. Where is this file coming from?
I don't see that file anywhere in the directory, or the entire package.

Comment (anjneeksharma) [reply packages/feedback-spa/src/app/modules/feedback/home/feedback-home.component.ts:2]: That file is there in my directory structure see bellow screenshot.
image

Comment (anjneeksharma) [reply packages/feedback-spa/src/app/modules/feedback/home/feedback-home.component.ts:2]: Bellow is the working screenshot.

image

Comment (deshmukhmayur) [reply packages/feedback-spa/src/app/modules/feedback/home/feedback-home.component.ts:2]: Where is that data coming from?

Comment (anjneeksharma) [reply packages/feedback-spa/src/app/modules/feedback/home/feedback-home.component.ts]: @deshmukhmayur  i updated directory structure and and also move mock file.
Please have a another look.

Comment (anjneeksharma) [reply packages/feedback-spa/src/app/modules/feedback/home/feedback-home.component.ts]: For now testing purpose i am using one-portal backend coz for one platform backend is not ready yet.

File: packages/feedback-spa/package.json [hunk 9]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "feedback",
+  "name": "feedbackspa",
```
Comment (sayak-sarkar) [inline packages/feedback-spa/package.json:2]:
```suggestion
  "name": "feedback-spa",
```

File: packages/feedback-spa/package.json (hunk 7 above)
Comment (riginoommen) [reply packages/feedback-spa/package.json]: Can you please rebase your code and push because your changes will replace the updated version

**Comment by riginoommen** [comment]
@anjneeksharma Please rebase your code with the master because your package dependency will get downgraded 

https://github.com/1-Platform/one-platform/pull/147/files/3b97a9e36778fa3ce5881ba873f7758d5aa0ff3c#r415818436

File: packages/feedback-spa/src/app/app.module.ts [hunk 10]
```diff
@@ -10,11 +10,11 @@ import { HttpClientModule } from '@angular/common/http';
 import { Apollo } from 'apollo-angular';
 import { HttpLink } from 'apollo-angular-link-http';
 import { AppService } from './app.service';
-import {FeedbackService} from './modules/feedback/feedback.service';
-import { FeedbackHomeComponent } from './modules/feedback//home/feedback-home.component';
-import {FeedbackItemComponent} from './modules/feedback/feedback-item/feedback-item.component';
-import { CountBoxComponent } from './modules/feedback/count-box/count-box.component';
-import { DropdownComponent } from './modules/feedback/dropdown/dropdown.component';
+import {FeedbackService} from './feedback/feedback.service';
+import { FeedbackHomeComponent } from './feedback//home/feedback-home.component';
```
[the remark this answers is not in the document]
Comment (sayak-sarkar) [reply packages/feedback-spa/src/app/app.module.ts:14]: I see that this change is still pending.

File: packages/feedback-spa/package.json (hunk 7 above)
Comment (anjneeksharma) [reply packages/feedback-spa/package.json]: Rebased.

image

File: packages/feedback-spa/src/app/app.module.ts (hunk 10 above)
[the remark this answers is not in the document]
Comment (anjneeksharma) [reply packages/feedback-spa/src/app/app.module.ts]: @sayak-sarkar Rabased/updated code.
